### PR TITLE
Add GraphQl gem and introduce board query

### PR DESCRIPTION
**Pull request by jalapl**
We have added graphql-ruby gem and introduced a query to fetch a single board type.

### Related issues
- This PR resolves #ISSUE_NUMBER

### Description

_Describe the changes and motivations for the pull request, unless obvious from the title._

### Todo

- [ ] _Insert todo_

### Review

- [ ] Check screenshots provided in the pull request
- [ ] Manual test №1 by _@replace_yourself_
- [ ] Manual test №2 by _@replace_yourself_

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that feature branch is up-to-date with `master` (if not - rebase it).
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.

### Screenshots

| Before                                        | After                                         |
| --------------------------------------------- | --------------------------------------------- |
| _Insert screenshots and/or screen recordings_ | _Insert screenshots and/or screen recordings_ |

| Screenshot < 1300px screen size               | Screenshot < 900px screen size                |
| --------------------------------------------- | --------------------------------------------- |
|            _Insert screenshot_                |             _Insert screenshot_               |

### CODE REVIEW

File: app/graphql/types/board_type.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Types
+  class BoardType < Types::BaseObject
```
Comment (akaoss) [inline app/graphql/types/board_type.rb:2]: Don't we need an entity?

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    description "The query root of this schema"
+
+    field :board, BoardType, null: true do
+      description "Find a board by ID"
+      argument :id, ID, required: true
+    end
+
+    def board(id:)
+      Board.find(id)
+    end
```
Comment (akaoss) [inline app/graphql/types/query_type.rb:12]: Or have you decided to use methods inside type files instead of entities?

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+  if Rails.env.development?
+    mount GraphiQL::Rails::Engine, at: "/graphiql", graphql_path: "/graphql"
```
Comment (akaoss) [inline config/routes.rb:3]: Typo:
```suggestion
    mount GraphiQL::Rails::Engine, at: "/graphql", graphql_path: "/graphql"
```

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,9 @@
...
+  end
+
+  post "/graphql", to: "graphql#execute"
```
Comment (akaoss) [inline config/routes.rb:6]: Why do we need different things for production and development?

File: app/graphql/types/query_type.rb (hunk 2 above)
Comment (jalapl) [reply app/graphql/types/query_type.rb:12]: I wanted to go for the fastest solution

File: config/routes.rb (hunk 3 above)
Comment (jalapl) [reply config/routes.rb:3]: it's ok. 

Graphiql is a web interface to test your queries, you can read about it here i.e: https://evilmartians.com/chronicles/graphql-on-rails-1-from-zero-to-the-first-query

File: config/routes.rb (hunk 4 above)
Comment (jalapl) [reply config/routes.rb:6]: It's been automatically generated, but my guess is, you don't want to expose graphiql query editor on production.

Comment (akaoss) [reply config/routes.rb:6]: Yes, now I got it, thanks.

**Pull request merged**